src/main/java/com/airbnb/airpal/modules/AirpalModule.java

```diff
@@ -270,12 +272,40 @@ public AWSCredentials provideAWSCredentials()
     @Provides
     public AmazonS3 provideAmazonS3Client(AWSCredentials awsCredentials)
     {
+        // build an encryption materials provider object to handle encryption key management if needed
+        EncryptionMaterialsProvider emp = null;
+        String empClassName = config.getS3EncryptionMaterialsProvider();
+        if (empClassName != null) {
+            try {
+                Class<?> empClass = Class.forName(empClassName);
+                Object instance = empClass.newInstance();
+                if (instance instanceof EncryptionMaterialsProvider) {
+                    emp = (EncryptionMaterialsProvider)instance;
+                }
+                else {
+                    throw new IllegalArgumentException("Class " + empClassName + " must implement EncryptionMaterialsProvider");
+                }
+            }
+            catch (Exception x) {
+                throw new RuntimeException("Unable to initialize EncryptionMaterialsProvider class " + empClassName + ": " + x, x);
+            }
+        }
```

Comment:
I think we want to move the `EncryptionMaterialsProvider` into its own provider method, having this method depend on it.